### DISCUSSION TITLE
subscribeState: Wie kann man abfragen ,ob ein state schon subcribed wurde?

### DISCUSSION THREAD

**Discussion by mcuiobroker** [Zefau/ioBroker.jarvis#2239] [Q&A]
DP wurde schon mit `window.Socket.subscribeState(...);` aufgenommen. 
Wie kann ich das jetzt feststellen?
Welcher Befehl muss dafür verwendet werden? Danke.
Kann man überhaupt einen DP mehrfach subscriben?

**Comment by Zefau** [comment]
>  Kann man überhaupt einen DP mehrfach subscriben?

Ja, sonst würde jarvis auch gar nicht funktionieren.

Was hast du denn vor?

**Comment by mcuiobroker** [comment]
Ich hatte im Log bei einigen `subscribed` DPs mehrfach eine Anzeige, obwohl ich nur einmal `subcribed` habe. Deswegen hatte ich gedacht, es muss doch eine Möglichkeit geben zu prüfen, ob man diesen DP schon `subscribed` hat?
Grundsätzlich scheint bei einem subscribeState 3x eine Anzeige zu geben, wenn man z.B. im
```
window.subscribeState('0_iserdata.0.condition',({err,state}) => {
  console.log(state.val);
})
```
den Wert ausgibt.
Hab es nochmal mit v3.1.3-beta.40 probiert, da wird es nur einmal ausgegeben. 
```

window.Socket.subscribeState('jarvis.'+jarvisInstance+'.settings.themeDarkMode', ({ err, state }) => {
	console.log('dark Mode: '+ state.val);
});
```
Ich muss es nochmal überarbeiten.
